### PR TITLE
add: email validation and confirm password checks

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
Added email pattern validation using regex: "[a-z0-9]+@[a-z]+.[a-z]{2,3}" and added confirm password check.
@polymahh @salahlalami

### CODE REVIEW

**Comment by polymahh** [comment]
thank you @vinaybadgujar102 
maybe a a cleaner solution we can use type email
image

**Comment by vinaybadgujar102** [comment]
@polymahh made change as you mentioned.

File: frontend/src/forms/RegisterForm.jsx [hunk 1]
```diff
@@ -58,7 +62,16 @@ export default function RegisterForm() {
             required: true,
             message: 'Enter Password Again!',
           },
+          ({ getFieldValue }) => ({
+            validator(_, value) {
+              if (!value || getFieldValue('password') === value) {
+                return Promise.resolve();
+              }
+              return Promise.reject(new Error('The new password that you entered do not match!'));
```
Comment (polymahh) [inline frontend/src/forms/RegisterForm.jsx:70]: change it to this new Error('The two passwords that you entered do not match!')

**Review by polymahh — CHANGES_REQUESTED** [review]

File: frontend/src/forms/RegisterForm.jsx (hunk 1 above)
Comment (vinaybadgujar102) [reply frontend/src/forms/RegisterForm.jsx]: made the requested changes

**Review by polymahh — APPROVED** [review]

**Review by salahlalami — APPROVED** [review]

**Pull request merged**